packages/common/src/streams.ts

```diff
@@ -107,9 +107,13 @@ export function decodeStream(
  * Create a series of decoding streams based on the content-encoding header. The
  * resulting streams should be piped together to decode the content.
  */
-export function createDecoders(contentEncoding?: string): Duplex[] {
+export function createDecoders(contentEncoding?: string | string[]): Duplex[] {
   const decoders: Duplex[] = []
 
+  if (Array.isArray(contentEncoding)) {
+    throw new TypeError('content-encoding must be a comma separated string')
+  }
```

Comment:
Not sure I totally understand this— why do we permit an array in the types if it's unsupported?

Reply:
Oh, is this something about representing duplicate headers?  If so, technically I think the move is supposed to be to combine them per https://datatracker.ietf.org/doc/html/rfc2616#section-4.2.

Reply:
NodeJS's http server will ensure that this header is never an array:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e58b2667d22cf8a4ba8a34c2b811b3e911c55583/types/node/http.d.ts#L69C1-L69C49

But Undici doesn't:
https://github.com/nodejs/undici/blob/fbdcdb5fd1ed69a0b8a3fb8444ff0e2ac9755e63/types/header.d.ts#L6

Because of node's implementation I assumed this was not allowed.

I adapted to allow arrays.